heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/V1Controller.java

```diff
@@ -130,6 +137,18 @@ boolean submit(PackingPlan packingPlan) {
       throw new TopologySubmissionException(e.getMessage());
     }
 
+    // Get and then create Persistent Volume Claims from the CLI.
+    persistentVolumeClaimConfigs = KubernetesContext.getPersistentVolumeClaims(getConfiguration());
+    if (KubernetesContext.getPersistentVolumeClaimDisabled(getConfiguration())
+        && !persistentVolumeClaimConfigs.isEmpty()) {
+      final String message =
+          String.format("Loading Persistent Volume Claim from CLI is disabled: '%s'", topologyName);
```

Comment:
This error message is not very clear. From the if statement it looks like when the PVC configs exist but PVC is disabled, the topology cannot be submitted? Also, Is it the expected behavior?

Also, it seems like you have a check of `persistentVolumeClaimConfigs != null` later. Do we need to check it here?

Reply:
Thank you @nwangtw for taking a look at this! 😄 

> This error message is not very clear. From the if statement it looks like when the PVC configs exist but PVC is disabled, the topology cannot be submitted? Also, Is it the expected behavior?

You are correct, if a PVC is provided on the CLI but the option is disabled there will be a hard failure. This is so there are no unwelcome surprises in the event a topology is expecting for there to be a PVC.

I will try to reword the error message to make it clearer.
 
> Also, it seems like you have a check of `persistentVolumeClaimConfigs != null` later. Do we need to check it here?

Good catch! I need to be more consistent. After `getPersistentVolumeClaims` is called the data structure will either be empty or fully populated. This was by design to avoid dealing with `nullptr` issues. I shall affect changes.

Reply:
Your confusion with the `getPersistentVolumeClaims` is case and point for why Javadoc comments are essential. I have been lazy because the file does not have any of them. I shall add them to the routines I have created.